### PR TITLE
Nelson adapt cabin sorting endpoints

### LINKED ISSUE #135: [Feature] Adapt Cabin Sorting Endpoints
- [ ] Once #134 is merged, adapt the cabin sorting project to use the Mongo connection in app.ts to handle question/answer fetching
- [ ] Create a routes directory
- [ ] Using express in a `sortedHackers-routes.ts` file, create a route "/sortedHackers" that returns a list of sorted hackers in JSON format with your existing logic

Review by @mike10911 @surbhi-gulati @alyssamui

### PR DESCRIPTION
**Pull request by NelsonDong41**
Closes #135 

Allows for posting and getting cabin sorting data from database endpoint.
Running script grabs data from endpoint and uses existing script to parse and sort hackers into cabins
Allows for get and post request for hackers through /sortedHackers route

### CODE REVIEW

File: backend/app.ts
Comment (alyssamui) [inline backend/app.ts:1]: is the { Application} import used?

File: backend/cabin-sorting/hackerSortingAlgo.ts
Comment (alyssamui) [inline backend/cabin-sorting/hackerSortingAlgo.ts:35]: can we put the URL in an .env variable instead so it isn't hardcoded to localhost:4000

File: backend/cabin-sorting/hackerSortingAlgo.ts
Comment (alyssamui) [inline backend/cabin-sorting/hackerSortingAlgo.ts:37]: is this console log for debugging or does it output useful information for the user/should this be parsed to be more user friendly

File: backend/cabin-sorting/schemas/sortedHackers-schema.ts
Comment (alyssamui) [inline backend/cabin-sorting/schemas/sortedHackers-schema.ts:3]: maybe move these enums to types.ts

File: backend/cabin-sorting/types.ts
Comment (alyssamui) [inline backend/cabin-sorting/types.ts:15]: nit: delete extra whitespace

File: backend/cabin-sorting/types.ts
Comment (alyssamui) [inline backend/cabin-sorting/types.ts:4]: if you move the enums from above down into this file, you can set the type to the corresponding enum instead of hard coding

**Review by alyssamui — CHANGES_REQUESTED** [review]

File: backend/cabin-sorting/controllers/sortedHackers-controller.ts
Comment (mike10911) [inline backend/cabin-sorting/controllers/sortedHackers-controller.ts:12]: Lets have a more descriptive variable name

File: backend/cabin-sorting/hackerSortingAlgo.ts
Comment (mike10911) [inline backend/cabin-sorting/hackerSortingAlgo.ts:35]: Can we type this?

File: backend/cabin-sorting/service/sortedHackers-service.ts
Comment (mike10911) [inline backend/cabin-sorting/service/sortedHackers-service.ts:20]: better variable name here as well

File: backend/cabin-sorting/service/sortedHackers-service.ts
Comment (mike10911) [inline backend/cabin-sorting/service/sortedHackers-service.ts:17]: I think we can clean this method up! Try using the map to transform the hackerData objects and reduce() instead of a for each loop to update the responses.

**Review by mike10911 — CHANGES_REQUESTED** [review]
Wow, you killed it Nelson!!!! Just a few comments below!

**Review by alyssamui — APPROVED** [review]

**Review by mike10911 — APPROVED** [review]
LGTM after merge conflicts are resolved!

**Review by surbhi-gulati — APPROVED** [review]

**Pull request merged**
